src/lua/LuaObject.h

```diff
@@ -376,6 +405,14 @@ inline void LuaObject<T>::PushToLua(const T &o)
 	Register(new (LuaObjectBase::Allocate(sizeof(LuaCopyObject<T>))) LuaCopyObject<T>(o));
 }
 
+template <typename T>
+template <typename... Args>
+inline void LuaObject<T>::CreateInLua(Args &&... args)
+{
+	T *p(new T(std::forward<Args>(args)...));
+	Register(new (LuaObjectBase::Allocate(sizeof(LuaOwnObject<T>))) LuaOwnObject<T>(static_cast<T *>(p)));
+}
+
```

Comment:
I would *strongly* recommend a comment on this function warning users that `CreateInLua` cannot be mixed with ref-counted pointers; when the Lua object is GC'd, it will delete the underlying object regardless if it is of a refcounted type and has active references pointing to it.

Reply:
@Web-eWorks but only the parameters of the constructor are passed to this function, so it is impossible to simply pass refcounted there, only through the copy-constructor maybe, and if I'm not mistaken, this will lead to an increase in the counter, and everything will be fine?

Reply:
However - the object that is generated in lua can be copied around perpetually and passed to other functions that might take a pointer to an object. Our calling convention is to `LuaPull<ClassName *>()`, not `LuaPull<RefCountedPtr<ClassName>>()`, which means that if CreateInLua is called for an object inheriting from `RefCounted` it's possible that it's pushed to Lua as a LuaOwnedObject, pulled as a `ClassName *`, and passed to a function expecting a `RefCountedPtr<ClassName>` (even if such function doesn't *yet* exist). In this case, either there will be a dangling reference or the object stored in LuaOwnObject will be deleted by a ref-count decrement.

I'd specifically like to see a comment that makes this restriction explicit so that a future junior programmer / contributor doesn't need to fully understand the nuances of the lua binding model to be able to make an informed decision about which of the *four* seemingly-equal methods of pushing data to Lua is viable for their specific situation.

Reply:
@Web-eWorks wrote a comment, hopefully ok